pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -293,11 +295,11 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration, featur
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: podLogsDir was not specified"))
 	}
 
-	if !filepath.IsAbs(kc.PodLogsDir) {
+	if !utilfs.IsAbs(kc.PodLogsDir) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: pod logs path %q must be absolute path", kc.PodLogsDir))
 	}
 
-	if filepath.Clean(kc.PodLogsDir) != kc.PodLogsDir {
+	if runtime.GOOS != "windows" && filepath.Clean(kc.PodLogsDir) != kc.PodLogsDir {
```

Comment:
hm, I assume we're skipping this because ``filepath.Clean`` replaces ``/`` with ``\``, so this check would fail. But shouldn't we do this check in some form? The check is meant to catch cases in which the configuration is wrong, so there may be some value in doing it.

Reply:
agreed

Reply:
Looked a bit into this. What about this:

```
filepath.ToSlash(filepath.Clean(path)) == filepath.ToSlash(path)
```

I've tested this myself on a Windows node:

```
package main

import (
        "fmt"
        "path/filepath"
)

func testCleanPath(path string) bool {
        // Clean will replace slashes to Separator (which is OS-specific).
        // We're making sure that all slashes are the same before comparing.
        return filepath.ToSlash(filepath.Clean(path)) == filepath.ToSlash(path)
}

func main() {
        cases := []struct {
                path     string
                expected bool
        }{
                {
                        path:     "/var/lib/something",
                        expected: true,
                },
                {
                        path:     "C:/var/lib/something",
                        expected: true,
                },
                {
                        path:     "C:\\var\\lib\\something",
                        expected: true,
                },
                {
                        path:     "var/lib/something",
                        expected: true,
                },
                {
                        path:     "var\\lib\\something",
                        expected: true,
                },
                {
                        path:     "/",
                        expected: true,
                },
                {
                        path:     "\\",
                        expected: true,
                },
                {
                        path:     "C:/",
                        expected: true,
                },
                {
                        path:     "C:\\",
                        expected: true,
                },
                {
                        path:     ".",
                        expected: true,
                },
                {
                        path:     "/var/../something",
                        expected: false,
                },
                {
                        path:     "C:\\var\\..\\something",
                        expected: false,
                },
                {
                        path:     "/var//lib/something",
                        expected: false,
                },
                {
                        path:     "C:/var//lib/something",
                        expected: false,
                },
                {
                        path:     "\\var\\\\lib\\something",
                        expected: false,
                },
                {
                        path:     "C:\\var\\\\lib\\something",
                        expected: false,
                },
                {
                        path:     "/var/./lib/something",
                        expected: false,
                },
                {
                        path:     "C:/var/./lib/something",
                        expected: false,
                },
                {
                        path:     "\\var\\.\\lib\\something",
                        expected: false,
                },
                {
                        path:     "C:\\var\\.\\lib\\something",
                        expected: false,
                },
        }

        errorCount := 0
        for _, tc := range cases {
                actual := testCleanPath(tc.path)
                if actual != tc.expected {
                        fmt.Printf("Actual: %q, expected: %q, for path: %s\n", actual, tc.expected, tc.path)
                        errorCount++
                }
        }

        fmt.Printf("Encountered %d errors.\n", errorCount)
}
```

Output from running it:

```
PS C:\Users\Administrator\workdir\kubernetes> go run .\tester.go
Encountered 0 errors.
```

Reply:
that seems reasonable to me

Reply:
@mxpv what are your thoughts on @claudiubelu's suggestion? Just want to call out that this issue is a blocker for RC2 and the release, so we need to reach a resolution and have this merged soon.

Reply:
I think this makes sense as well. Perhaps adding a IsCleanPath(str) function into utilfs?

Reply:
@aravindhp sure, I will update the PR shortly.